src/dune_engine/build_config.ml

```diff
@@ -96,13 +96,15 @@ type t =
   ; execution_parameters : dir:Path.Source.t -> Execution_parameters.t Memo.t
   ; source_tree : (module Source_tree)
   ; action_runner : Action_exec.input -> Action_runner.t option
+  ; action_runners : unit -> Action_runner.t list
```

Comment:
What's the difference between this and the field above?

Reply:
First one is for selecting the appropriate action runner based on the actions, second one is for notifying all action runners. Merging those is non-trivial since selection depends on some JS-specific code. We plan to refactor this but for now we're trying to minimize the diff between our version and upstream.

Reply:
Very well. Can you include a comment in the mli that says this will be refactored at a later date? Descriptions of the fields would also be nice but not necessary.